dask/array/core.py

```diff
@@ -1002,20 +969,9 @@ def _add_blockwise_layer_for_keyword(keyword_arr, prefix):
                 ),
                 "dtype": dtype,
             }
-            block_info_arr[block_id] = info
+            block_info_dict[block_id] = info
 
-        cs = tuple((1,) * len(c) for c in out.chunks)
-        dsk = graph_from_arraylike(
-            block_info_arr,
-            cs,
-            block_info_arr.shape,
-            block_info_name,
-            getitem=_getter_item,
-            dtype=block_info_arr.dtype,
-            inline_array=False,
-        )
```

Comment:
`graph_from_arraylike` isn't used anywhere (it's imported in dask-expr, we might want to nuke the array modules there to simplify work)